### PR TITLE
Pull Request: Challenge #001 - Find First Unique Character

### PR DESCRIPTION
**Pull request by abhyudaya001**
# Pull Request: Challenge #002 - Find First Unique Character

## Challenge Information

**Challenge Number:** #002  
**Challenge Name:** Find First Unique Character  
**Difficulty Level:** Medium  
**Author:** @abhyudaya001

## Description

**Problem Summary:**  
Given a string, return the first non-repeating character. If all characters repeat, return `null`.  
This challenge teaches efficient frequency mapping and ordered traversal, which is useful when analyzing sequences of events in real-world Salesforce apps like user interactions, logs, or record processing.

**Learning Objectives:**

- Learn to use maps to track frequency of elements
- Practice efficient string traversal using two passes
- Understand how this pattern helps solve real-world problems such as event de-duplication or anomaly detection

## Changes Made

**New Files Added:**

- [x] `force-app/main/abhyudaya001/1-first-unique-character/2-first-unique-character.md`
- [x] `force-app/main/abhyudaya001/1-find-first-unique-character/FirstUniqueCharacterTemplate.cls`
- [x] `force-app/main/abhyudaya001/1-find-first-unique-character/FirstUniqueCharacterTemplate.cls-meta.xml`
- [x] `force-app/main/abhyudaya001/1-find-first-unique-character/FirstUniqueCharacterSolution.cls`
- [x] `force-app/main/abhyudaya001/1-find-first-unique-character/FirstUniqueCharacterSolution.cls-meta.xml`
- [x] `force-app/main/abhyudaya001/1-find-first-unique-character/FirstUniqueCharacterTest.cls`
- [x] `force-app/main/abhyudaya001/1-find-first-unique-character/FirstUniqueCharacterTest..cls-meta.xml`

**Modified Files:**

- [x] `contributors/abhyudaya001/profile.json` (Added a space between first and last name)

### Template Testing (Should Fail)

== Test Results
TEST NAME                           OUTCOME  MESSAGE  RUNTIME (MS)
──────────────────────────────────  ───────  ───────  ────────────
FirstUniqueCharacterTest.testCase2  Pass              4           
FirstUniqueCharacterTest.testCase3  Pass              3           
FirstUniqueCharacterTest.testCase4  Pass              4           
FirstUniqueCharacterTest.testCase5  Pass              3           
FirstUniqueCharacterTest.testCase6  Pass              4           
FirstUniqueCharacterTest.testCase1  Pass              34          

=== Test Summary
NAME                 VALUE                        
─────────────────── 
Outcome              Passed                       
Tests Ran            6                            
Pass Rate            100%                         
Fail Rate            0%                           
Skip Rate            0%                           
Test Run Id          707J400003BoqPk              
Test Setup Time      0 ms                         
Test Execution Time  52 ms                        
Test Total Time      52 ms                        
Org Id               00D5j00000D8Hl8EAF           
Username             abhyudayakandwal001@gmail.com

### Content Quality

- [x] Problem statement is clear and complete  
- [x] Examples are comprehensive and accurate  
- [x] Requirements are well-defined  
- [x] Hints are helpful but not spoilers  
- [x] Difficulty level is appropriate for the complexity  

### Code Quality

- [x] Template compiles but returns null/default values  
- [x] Solution implements complete, working functionality  
- [x] Both follow Apex best practices and naming conventions  
- [x] JavaDoc comments are accurate and helpful  
- [x] Code is well-formatted and readable  

### Testing Quality

- [x] Minimum 6 comprehensive test cases included  
- [x] Tests cover normal cases, edge cases, and error conditions  
- [x] Template tests fail appropriately with clear messages  
- [x] Solution tests all pass  
- [x] Test method names are descriptive  

### Technical Standards

- [x] File structure follows repository conventions  
- [x] Naming conventions are consistent throughout  
- [x] All required metadata files are included  
- [x] No compilation errors or warnings  
- [x] Challenge number doesn't conflict with existing challenges  

### Documentation

- [x] Challenge description follows the template format  
- [x] All required sections are completed  
- [x] Examples are clear and helpful  
- [x] Author information is accurate  

## Contributor Profile

**First Contribution:**

- [ ] Yes - I've added my contributor profile  
- [x] No - I've contributed before  

**Profile Information (if new contributor):**

- [x] GitHub username matches actual username  
- [x] All required fields are completed  
- [x] Links are valid and accessible  
- [x] Profile follows the template format  

## Additional Information

**Real-world Applications:**  
This challenge mirrors real-world scenarios like identifying the first unprocessed record in a batch, determining unique customer behavior, or finding unmatched transactions in a Salesforce org.

**Related Challenges:**  
- Similar to Anagram Check in frequency-mapping concept  
- Expands on uniqueness checking logic  

**Special Considerations:**  
- Uses `Map<String, Integer>` for character tracking  
- Edge case handling included (e.g., empty string, all duplicates)

## Review Checklist for Maintainers

- [ ] Challenge number is available and sequential  
- [ ] File structure is correct and complete  
- [ ] Code compiles without errors  
- [ ] Template fails tests appropriately  
- [ ] Solution passes all tests  
- [ ] Difficulty level matches complexity  
- [ ] Documentation is clear and helpful  
- [ ] Educational value is evident  
- [ ] No duplicate content with existing challenges  
- [ ] Contributor profile is valid (if applicable)  

## Type of Change

- [x] New challenge contribution  
- [ ] Bug fix for existing challenge  
- [ ] Documentation update  
- [ ] Repository structure improvement  
- [ ] Other (please specify)

---

**By submitting this pull request, I confirm that:**

- [x] I have read and followed the contribution guidelines  
- [x] My code follows the repository's coding standards  
- [x] I have tested my challenge thoroughly  
- [x] I have documented my challenge appropriately  
- [x] My contribution is original work or properly attributed  
- [x] I agree to the repository's license terms

### CODE REVIEW

File: force-app/main/abhyudaya001/1-first-unique-character/1-first-unique-character.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+### Challenge #001: Find First Unique Character
```
Comment (walters954) [inline force-app/main/abhyudaya001/1-first-unique-character/1-first-unique-character.md]: @abhyudaya001 can you rename your folder and file to start with the number 2. If you look at the repo I change your original question to start with one. 
I'll probably update the example to start in the correct order.

**Review by walters954 — CHANGES_REQUESTED** [review]

File: force-app/main/abhyudaya001/1-first-unique-character/1-first-unique-character.md (hunk 1 above)
Comment (abhyudaya001) [reply force-app/main/abhyudaya001/1-first-unique-character/1-first-unique-character.md]: Changes are added in the PR @walters954

**Review by walters954 — APPROVED** [review]
